tools/jinja2-filters.py

```diff
@@ -18,3 +24,24 @@ def filter_b64decode(encoded_value, return_type = "string"):
         return b64decode(encoded_value, validate = True)
     else:
         raise UnsupportedDataType("The encoded value is an unsupported data type")
+
+def filter_chained_get(dictionary, key_list):
+    if not isinstance(dictionary, dict):
+        raise NotADict("The variable is not a dict")
+    if not isinstance(key_list, str):
+        raise NotAString("The chain segment is not a string")
+
+    chain_segments = key_list.split('.')
+    value = dictionary
+    for chain_segment in chain_segments:
+        if isinstance(value, dict) and chain_segment in value:
+            if value is None:
+                return value
```

Comment:
This will never be true because if you're here, `value` must be a `dict`, not `None`.